File: JavaScript/authorization.js
const request = require('request')

// Reminder to add 'token ' before auth from Hub
HUB_CONFIG = {
    host: 'https://hubtest.com',
    token: 'token '
}
const headers = {
    'authorization': HUB_CONFIG.token,
    'cache-control': 'no-cache'
}

const options = {
    url: HUB_CONFIG.host + '/api/tokens/authenticate',
    method: 'POST',
    headers: headers
}

function authorizeHub() {
    request(options, function (error, response, body) {
        if (error) {
            return console.log(error)
        }
        
        let res = JSON.parse(body)
        // Pass bearer token as auth headers during subsequent requests
        return console.log(res.bearerToken)

    })
}

authorizeHub()

File: Python/authorization.py
import http.client
import json

# Hub Variables
# Reminder to add 'token ' before the auth token from the Hub

HUB_CONFIG = {
    'host': 'hub.com',
    'token': 'token '
}

# Hub Connect

class HubAuthorization:
    def authorizeHub(self):
        try:
            hubConn = http.client.HTTPSConnection(HUB_CONFIG['host'])
            headers = {
                'authorization': HUB_CONFIG['token'],
                'cache-control': 'no-cache'
            }
            hubConn.request('POST', '/api/tokens/authenticate', headers=headers)
            res = hubConn.getresponse()
            data = res.read().decode('utf-8')
            auth_obj = json.loads(data)
            # Pass bearer token as auth headers in subsequent requests
            bearerToken = auth_obj['bearerToken']
            print('Hub Authorization Succeeded.')
            return bearerToken

        except Exception as e:
            print('Cannot connect to Hub. Invalid token.')
            print(e)



#authorize = HubAuthorization()
#authorize.authorizeHub()